projects/packages/stats-admin/src/class-admin-post-list-column.php

```diff
@@ -0,0 +1,245 @@
+		 *
+		 * If the Odyssey Stats experience isn't enabled, the user will need to go to Calypso,
+		 * so they need to be connected to WordPress.com to be able to access that page.
+		 */
+		if (
+			! current_user_can( 'view_stats' )
+			|| (
+				! Stats_Options::get_option( 'enable_odyssey_stats' )
+				&& ! ( new Connection_Manager( 'jetpack' ) )->is_user_connected()
+			)
+		) {
+			return $columns;
+		}
+
+		// Array-Fu to add before comments.
+		$pos = array_search( 'comments', array_keys( $columns ), true );
+
+		// Fallback to the last position if the post type does not support comments.
+		if ( ! is_int( $pos ) ) {
+			$pos = count( $columns );
+		}
+
+		$chunks             = array_chunk( $columns, $pos, true );
+		$chunks[0]['stats'] = esc_html__( 'Stats', 'jetpack-stats-admin' );
+
+		return call_user_func_array( 'array_merge', $chunks );
+	}
+
+	/**
+	 * Get a list of post views for each post id from the global $wp_query.
+	 *
+	 * @return array
+	 */
+	public function get_post_page_views_for_current_list(): array {
+		global $wp_query;
+
+		if ( ! $wp_query->posts ) {
+			return array();
+		}
+
+		$post_ids = wp_list_pluck( $wp_query->posts, 'ID' );
+
+		$wpcom_stats = $this->get_stats();
+		$post_views  = $wpcom_stats->get_total_post_views( array( 'post_ids' => implode( ',', $post_ids ) ) );
+
+		if ( is_wp_error( $post_views ) || empty( $post_views ) ) {
+			return array();
+		}
+
+		$views = array();
+
+		foreach ( $post_views['posts'] as $post ) {
+			$views[ $post['ID'] ] = $post['views'];
+		}
+
+		return $views;
+	}
+
+	/**
+	 * Get the stats object.
+	 *
+	 * @return WPCOM_Stats
+	 */
+	protected function get_stats() {
+		return new WPCOM_Stats();
+	}
+
+	/**
+	 * Get the NumberFormatter instance.
+	 *
+	 * @param string $locale The current locale.
+	 *
+	 * @return NumberFormatter
+	 */
+	protected function get_formatter( string $locale ): \NumberFormatter {
+		if ( isset( $this->formatter[ $locale ] ) ) {
+			return $this->formatter[ $locale ];
+		}
+
+		/**
+		 * PHP's NumberFormatter is just a wrapper over the ICU C library. The library does support decimal compact short formatter, but PHP doesn't have a stub for it (=< PHP 8.4).
+		 *
+		 * @see https://unicode-org.github.io/icu-docs/apidoc/dev/icu4c/unum_8h.html UNUM_DECIMAL_COMPACT_SHORT constant.
+		 */
+		$compact_decimal_short = 14;
+
+		/**
+		 * NumberFormatter::DECIMAL_COMPACT_SHORT only exists in PHP 8.5 and later. At this time, NumberFormatter::DECIMAL_COMPACT_SHORT only exists in PHP `main` branch.
+		 *
+		 * Use the constant if it's defined since it's safer.
+		 */
+		if ( defined( '\NumberFormatter::DECIMAL_COMPACT_SHORT' ) ) {
+			// @phan-suppress-next-line PhanUndeclaredConstantOfClass
+			$compact_decimal_short = NumberFormatter::DECIMAL_COMPACT_SHORT;
+		}
+
+		try {
+			$formatter = new \NumberFormatter( $locale, $compact_decimal_short );
+			$formatter->setAttribute( \NumberFormatter::MAX_FRACTION_DIGITS, 1 );
+		} catch ( \Exception $e ) {
```

Comment:
As pointed out here: paYJgx-5QS-p2#comment-5786 [Jetpack doesn't require the `intl` extension](https://jetpack.com/support/getting-started-with-jetpack/), this will need a further fallback...Maybe just ducktype something that hardcodes SI suffixes?